src/UserAuthenticationServiceProvider.php

```diff
@@ -68,6 +68,11 @@ public function boot()
             }
         }
 
+        // check for email whitelist/blacklist and not both
+        if (config('user-authentication.email_restrictions.mode') === 'both') {
+            \Log::warning('UserAuthentication: Both email whitelist and blacklist are set. Please configure only one.');
```

Comment:
Why check for `both`? in the config file you specified 3 possible values and `both` is not among the 3 values you specified.